src/main/java/io/jenkins/plugins/neuvector/NeuVectorBuilder.java

```diff
@@ -859,13 +874,23 @@ protected Object readResolve() {
                         if (migratedFromNonTLSVersion) {
                             disableTLSCertVerification = (user != null && !user.isEmpty());
                         }
-                        controllerEndpointUrls.add(new ControllerEndpointUrl(defaultNickname, controllerApiUrl, user, password, disableTLSCertVerification, serverCertificate, timeout));
+                        disableAPIKeyVerification = (serverApiKey == null);
+                        controllerEndpointUrls.add(new ControllerEndpointUrl(defaultNickname, controllerApiUrl, user, password, disableTLSCertVerification, disableAPIKeyVerification, serverCertificate, timeout, serverApiKey));
                     } catch (URISyntaxException e) {
                         e.printStackTrace();
                     }
                     isDeafultControllerEndpointUrls = true;
                 }
+            } else {
+                // Case Upgrade from 2.4, make sure disableAPIKeyVerification are set as true if user has not configured it
+                for (ControllerEndpointUrl controllerEndpointUrl : controllerEndpointUrls) {
+                    // if user has not configured disableAPIKeyVerification, set it as true
+                    if (controllerEndpointUrl.getDisableAPIKeyVerification() == null) {
```

Comment:
getDisableAPIKeyVerification() returns boolean. Can a boolean var be null in Java ?
Why not call getServerApiKey() instead?

Reply:
Hi @williamlin-suse 
I used a Boolean here, as it can be null. A null value represents that it has never been set before, which applies to the update case.

Alternatively, getServerApiKey() also works.

Which approach do you prefer?